pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationsAPI.kt

```diff
@@ -104,6 +104,11 @@ class PushNotificationsAPI(private val instanceId: String, overrideHostURL: Stri
   // Handles the JsonSyntaxException properly
   private fun safeExtractJsonError(possiblyJson: String): NOKResponse {
     return try {
+      // This if check is added to make it work as old Kotlin version.
+      // Need to be revisited in future
```

Comment:
Is this a WIP change in this branch, or does "revisited in the future" refer to some tech-debt that is being knowingly introduced now as a workaround? If it's the latter, then can you please track this by opening an issue discussing how this might be resolved later?

Reply:
I am not sure if this creates an issue. Without this change the API throws RuntimeException with new version of Kotlin. Honestly,I think I can remove this comment.